acra/acra-in-depth/data-structures/_index.md

```diff
@@ -2,3 +2,185 @@
+
+| Email        | Column 2     | Column 3     |
+|--------------|--------------|--------------|
+| <AcraStruct> | Column2Value | Column3Value |
+| <AcraStruct> | Column2Value | Column3Value |
+
+In this case, an AcraStruct takes up a whole cell and we are trying to decrypt it as is, without searching for anything.
+But things can be different. In InjectedCell mode a binary [MsgPack](https://msgpack.org/index.html) or [protobuf](https://developers.google.com/protocol-buffers/) could be stored in a table, and it is possible that in those data entities only one field is encrypted.
+
+Such data entity wouldn’t be a single AcraStruct - it would be a data entity that contains an AcraStruct or several AcraStructs. This means that in the InjectedCell mode we stop assuming that the database cells can only contain complete AcraStructs. AcraStructs can be inside some other pieces of data and that’s where we’re starting to look for them. This consequently slows down the processing speed. However, not every task needs the InjectedCell mode, which is why one can switch between modes, depending on what you’re encrypting and how you’re storing it.
+
+By default, the WholeCell mode is active (`--acrastruct_wholecell_enable`). You can switch between the modes using startup parameters:
+
+```
+--acrastruct_injectedcell_enable
+    Acrastruct may be injected into any place of data cell
+--acrastruct_wholecell_enable
+    Acrastruct will stored in whole data cell (default true)
+```
+
+Also see the [CLI reference](/pages/documentation-acra/#acraserver-cli-reference).
+
+## Acrastruct validator
+
+To decrypt AcraStructs generated by AcraWriter in your application, you can also use our [Acrastruct validator](/simulator/acra/). It works as AcraServer / AcraTranslator and uses private decryption keys.
+
+### Getting started with Acrastruct validator
+
+To use Acrastruct validator, you need to register (or login if you have already registered) using your email address (don't worry, no spam from us):
+
+**1.** Click "Register" in the top right corner of Documentation Server (and skip over to Step 3 if you've already registered).
+
+![](https://docs.cossacklabs.com/files/wiki/Cossack-Labs-Themis-Interactive-Simulator-Login.png)
+
+**2.** Register using your email address and create a password (we're feeling awkward reminding you about it, but you should use a unique strong password, always).
+
+![](https://docs.cossacklabs.com/files/wiki/Cossack-Labs-Themis-Interactive-Simulator-Register.png)
+
+**3.** Login using your password and email address.
+
+![](https://docs.cossacklabs.com/files/wiki/Cossack-Labs-Themis-Interactive-Simulator-Login-Screen.png)
+
+**4.** Go to the [Acrustruct validator tab in the Documentation server header](/simulator/acra/):
+
+![](https://docs.cossacklabs.com/files/wiki/AcrastructValidator-link.png)
```

Comment:
nice catch, I will try to find the correct link to image. Because there I had migrated content as is. So this image should not be rendered on our current docserver too...